.github/workflows/ci.yml

```
@@ -0,0 +1,90 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  qa:
+    name: QA
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toxenv:
+          - checkqa
+          - readme
+    steps:
+      - uses: actions/checkout@master
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+      - name: Set PY
+        run: echo "::set-env name=PY::$(python -VV | sha256sum | cut -d' ' -f1)"
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
```

Instead of setting a global env var, I'd rather use outputs:
```suggestion
      - name: Set PY
        id: py-ver
        run: echo "::set-output name=sha-256::$(python -VV | sha256sum | cut -d' ' -f1)"
      - uses: actions/cache@v1
        with:
          path: ~/.cache/pre-commit
          key: pre-commit|${{ jobs.py-ver.outputs.sha-256 }}|${{ hashFiles('.pre-commit-config.yaml') }}
```
_(https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter)_

---

It's taken from official setup https://pre-commit.com/#github-actions-example

---

I know, but when Anthony wrote that official setup, I think the commands were not fully documented. No need to pollute env now.

---

[]()

---

Have you tried to update official docs?

---

Not yet, I'm currently building a state-of-the-art workflow @ https://github.com/ansible/pylibssh/blob/devel/.github/workflows/build-test-n-publish.yml that tests the dists that are going to be published rather than testing the source separately. And when I'm done with polishing it I'll go and contribute the lessons learned to other places...

---

@asottile I wonder what do you think?

---

if it works send a patch 🤷 -- I don't use the github action all that much myself

---

Addressed in 988dfbc.

---

> if it works send a patch 🤷 -- I don't use the github action all that much myself

Addressed in https://github.com/pre-commit/pre-commit.com/pull/432